src/SqlUtils.ts

```diff
@@ -4,39 +4,54 @@ import * as mssql from "mssql";
 import Constants from "./Constants";
 import SqlConnectionConfig from "./SqlConnectionConfig";
 
+export interface ConnectionResult {
+    success: boolean,                   // True if connection succeeds, false otherwise
+    connection?: mssql.ConnectionPool,  // The connection object on success
+    error?: mssql.ConnectionError,      // Connection error on failure
+    ipAddress?: string                  // Client IP address if connection fails due to firewall rule
+}
+
 export default class SqlUtils {
 
     /**
      * Tries connection to server to determine if client IP address is restricted by the firewall.
      * First tries with master connection, and then with user DB if first one fails.
-     * Returns the client IP address if firewall restriction is present, or an empty string if connection succeeds. Throws otherwise.
+     * @param SqlConnectionConfig The connection configuration to try.
+     * @returns The client IP address if firewall restriction is present, or an empty string if connection succeeds. Throws otherwise.
      */
     static async detectIPAddress(connectionConfig: SqlConnectionConfig): Promise<string> {
         // First try connection to master
-        let ipAddress = await this.tryConnection(connectionConfig, true, true);
-        if (ipAddress !== undefined) {
-            return ipAddress;
+        let result = await this.tryConnection(connectionConfig, true);
+        if (result.success) {
+            result.connection?.close();
+            return '';
+        }
+        else if (result.ipAddress !== undefined) {
```

Comment:
this can be simplified to `else if (result.ipAddress)` if we don't have to worry about `result.ipAddress` being an empty string and still returning it when it is

Reply:
Updated